src/routes/search.js

```diff
@@ -95,21 +116,24 @@ class SearchRoute extends Component {
 
     // when details are not visible, we need to focus the last active
     // list item as determined by the `id` URL param
-    if (hideDetails && this.props.match.params.id !== match.params.id) {
-      shouldFocusItem = this.props.match.params.id || null;
+    if (hideDetails && match.params.id !== prevProps.match.params.id) {
+      shouldFocusItem = prevProps.match.params.id || null;
     }
 
-    // always update the results state
-    this.setState({
-      hideDetails,
-      shouldFocusItem,
-      searchType,
-      params,
-      sort: params.sort,
-      searchString: params.q,
-      searchFilter: params.filter,
-      searchField: params.searchfield
-    });
+    // update searchstring state only when it actually changes in the location instead of updating it each time on
+    // input to text field. This eliminates re-rendering of the text field on each keyboard in and solves problem
+    // stated in https://issues.folio.org/browse/UIEH-558
+    // The eslint disable line that we added in the line below can be removed after we move to React 17.0
+    if (!isEqual(location, prevProps.location)) {
+      this.setState({ searchString: params.q }); // eslint-disable-line react/no-did-update-set-state
+    }
+
+    // Rest of the state is updated in getDerivedStateFromProps except the one below whose state is
+    // computed after component updates. So, update it here.
+    // The eslint disable line that we added in the line below can be removed after we move to React 17.0
+    if (!isEqual(match, prevProps.match)) {
```

Comment:
Instead of checking the match, let's check if `shouldFocusItem !== this.state.shouldFocusItem`. That way it gets set back to `null` when needed, and it won't update if the `id` didn't change.